com.unity.shadergraph/Editor/Util/MessageManager.cs

```diff
@@ -143,48 +166,91 @@ void DebugPrint()
             Debug.Log(output.ToString());
         }
 
-        public static void Log(AbstractMaterialNode node, string path, ShaderMessage message, Object context)
+        public static void Log(AbstractMaterialNode node, string path, ShaderMessage message, Object context, IErrorLog log = null)
         {
+            if (log == null)
+                log = DebugErrorLog.Get();
+
             var errString = $"{message.severity} in Graph at {path} at node {node.name}: {message.message}";
             if (message.severity == ShaderCompilerMessageSeverity.Error)
             {
-                Debug.LogError(errString, context);
+                log.LogError(errString, context);
             }
             else
             {
-                Debug.LogWarning(errString, context);
+                log.LogWarning(errString, context);
             }
         }
 
-        public static void Log(string path, ShaderMessage message, Object context)
+        public static void Log(string path, ShaderMessage message, Object context, IErrorLog log = null)
         {
+            if (log == null)
+                log = DebugErrorLog.Get();
+
             var errString = $"{message.severity} in Graph at {path} on line {message.line}: {message.message}";
             if (message.severity == ShaderCompilerMessageSeverity.Error)
             {
-                Debug.LogError(errString, context);
+                log.LogError(errString, context);
             }
             else
             {
-                Debug.LogWarning(errString, context);
+                log.LogWarning(errString, context);
             }
         }
 
-        public bool AnyError()
+        public bool AnyError(Func<string, bool> nodeFilter = null)
```

Comment:
Yes, it would be pretty equivalent to:
return ErrorStrings(nodeFilter).Any();
only real downside is the enumerator does a GC alloc, whereas this straight loop does not.